GiEnJul/Controllers/RecipientController.cs

```diff
@@ -82,13 +86,39 @@ public async Task<ActionResult<(string FamilyId,string ReferenceId)>> PostAsync(
         [Authorize(Policy = Policy.ReadRecipient)]
         public async Task<List<Recipient>> GetRecipientsByInstitutionAsync([FromQuery] string institution)
         {
-        
             var recipients = await _recipientRepository.GetRecipientsByInstitutionAsync(institution);
-            foreach (var recipient in recipients)
-            {
-                recipient.FamilyMembers = await _personRepository.GetAllByRecipientId(recipient.RecipientId);
-            }
+            var recipientIds = recipients.Select(r => r.RecipientId);
+            var persons = await _personRepository.GetAllByRecipientIds(recipientIds);
+
+            recipients.ForEach(r => r.FamilyMembers = persons.Where(p => p.RecipientId == r.RecipientId).ToList());
+
             return recipients;
         }
+
+        [HttpGet("excel")]
+        [Authorize(Policy = Policy.ReadRecipient)]
+        public async Task<FileStreamResult> GetExcelListForInstitutionAsync([FromQuery] string institution)
```

Comment:
Can we be using the user metadata instead of sending in the institution?

Reply:
Probably. Will try it out when I do the rest of #787